Classes/Domain/Model/Product.php

```diff
@@ -982,11 +982,11 @@ public function updateAttributeValue(Attribute $attribute, $value): void
     /**
      * Return listing image if found.
      *
-     * @return Image|null
+     * @return Image[]|null
      */
-    public function getListImage(): ?Image
+    public function getListImages(): ?array
     {
-        return $this->findImageByType(Image::LISTING_IMAGE) ?? $this->getMainImage();
+        return $this->findImagesByType(Image::LISTING_IMAGE) ?? [$this->getMainImage()];
```

Comment:
I suggest this method always returns an array. Instead of returning null, return an empty array.